device/coordinate_manager.cpp

```diff
@@ -38,179 +38,53 @@ CoordinateManager::CoordinateManager(
+        case CoordSystem::LOGICAL: {
+            const tt_xy_pair physical_pair = to_physical_map[core_coord];
+            return CoreCoord(physical_pair.x, physical_pair.y, core_coord.core_type, CoordSystem::PHYSICAL);
+        }
     }
-
-    // Coord system is surely logical.
-    auto& logical_mapping = get_logical_to_physical(core_coord.core_type);
-    return logical_mapping[{core_coord.x, core_coord.y}];
 }
 
 CoreCoord CoordinateManager::to_virtual(const CoreCoord core_coord) {
     switch (core_coord.coord_system) {
-        case CoordSystem::TRANSLATED:
-        case CoordSystem::PHYSICAL:
-            return to_virtual(to_logical(core_coord));
         case CoordSystem::VIRTUAL:
             return core_coord;
+        case CoordSystem::PHYSICAL:
+            return from_physical_map[{{core_coord.x, core_coord.y}, CoordSystem::VIRTUAL}];
+        case CoordSystem::TRANSLATED:
+        case CoordSystem::LOGICAL:
+            return to_virtual(to_physical(core_coord));
     }
-
-    // Coord system is surely logical.
-    auto& logical_mapping = get_logical_to_virtual(core_coord.core_type);
-    return logical_mapping[{core_coord.x, core_coord.y}];
 }
 
 CoreCoord CoordinateManager::to_logical(const CoreCoord core_coord) {
     switch (core_coord.coord_system) {
         case CoordSystem::LOGICAL:
             return core_coord;
-        case CoordSystem::PHYSICAL: {
-            auto& physical_mapping = get_physical_to_logical(core_coord.core_type);
-            return physical_mapping[{core_coord.x, core_coord.y}];
-        }
-        case CoordSystem::VIRTUAL: {
-            auto& virtual_mapping = get_virtual_to_logical(core_coord.core_type);
-            return virtual_mapping[{core_coord.x, core_coord.y}];
-        }
-        case CoordSystem::TRANSLATED: {
-            auto& translated_mapping = get_translated_to_logical(core_coord.core_type);
-            return translated_mapping[{core_coord.x, core_coord.y}];
-        }
+        case CoordSystem::PHYSICAL:
+            return from_physical_map[{{core_coord.x, core_coord.y}, CoordSystem::LOGICAL}];
+        case CoordSystem::VIRTUAL:
+        case CoordSystem::TRANSLATED:
+            return to_logical(to_physical(core_coord));
     }
 }
 
 CoreCoord CoordinateManager::to_translated(const CoreCoord core_coord) {
     switch (core_coord.coord_system) {
         case CoordSystem::PHYSICAL:
+            return from_physical_map[{{core_coord.x, core_coord.y}, CoordSystem::TRANSLATED}];
+        case CoordSystem::LOGICAL:
         case CoordSystem::VIRTUAL:
-            return to_translated(to_logical(core_coord));
+            return to_translated(to_physical(core_coord));
         case CoordSystem::TRANSLATED:
             return core_coord;
     }
-
-    // Coord system is surely logical.
-    auto& logical_mapping = get_logical_to_translated(core_coord.core_type);
-    return logical_mapping[{core_coord.x, core_coord.y}];
```

Comment:
This comment refers to `CoreCoord CoordinateManager::to(const CoreCoord core_coord, const CoordSystem coord_system)` function....

I think you should write it like this:
```
CoreCoord CoordinateManager::to(const CoreCoord core_coord, const CoordSystem coord_system) {
    if (core_coord.coord_system == CoordSystem::PHYSICAL) {
        return from_physical_map.at({{core_coord.x, core_coord.y}, coord_system});
    } else {
        return to(to_physical_map.at(core_coord), coord_system);
        // You can also replace with next line if you don't like recursion:
        // return from_physical_map.at({to_physical_map.at(core_coord), coord_system});
    }
}
```
This way, you always check correctness of input coordinate... Again, if you want better error handling, you can use `std::map::find` instead of `std::map::at`...